### ISSUE TITLE
bitbucket cloud cannot query pull request status

### ISSUE THREAD

**Issue by jstrachan** [jenkins-x/jx#664]
I tried doing promotion to a bitbucket cloud based environment and the PR is created fine but it then fails querying the PR status:
```
+ jx promote -b --all-auto --timeout 1h --version 0.0.2
Promoting app demo5 version 0.0.2 to namespace jx-staging-bb2
Using pipeline: jstrachan/demo5/master build: #2
Cloning into '/home/jenkins/.jx/environments/jstrachan/environment-jx-staging-bb2'...
Found remote branch names master, HEAD -> origin/master, master
Switched to branch 'promote-demo5-0.0.2'
[promote-demo5-0.0.2 8d8c2af] Promote demo5 to version 0.0.2
 1 file changed, 8 insertions(+), 5 deletions(-)
remote: 
remote: Create pull request for promote-demo5-0.0.2:        
remote:   https://bitbucket.org/jstrachan/environment-jx-staging-bb2/pull-requests/new?source=promote-demo5-0.0.2&t=1        
remote: 
To https://bitbucket.org/jstrachan/environment-jx-staging-bb2.git
 * [new branch]      HEAD -> promote-demo5-0.0.2
Created Pull Request: https://bitbucket.org/jstrachan/environment-jx-staging-bb2/pull-requests/1

Using pipeline: jstrachan/demo5/master build: #2
error: Failed to query the Pull Request status for https://bitbucket.org/jstrachan/environment-jx-staging-bb2/pull-requests/1 Status: 404 Not Found, Body: {"type": "error", "error": {"message": "Resource not found", "detail": "There is no API hosted at this URL.\n\nFor information about our API's, please refer to the documentation at: https://developer.atlassian.com/bitbucket/api/2/reference/"}}
```

I wonder if there's a way to query the status for bitbucket? Maybe we have to use Jenkins itself to query the status if the bitbucket REST API doesn't support it?

**Comment by jstrachan** [comment]
Incidentally the Activity tab does show activity on a PR: 
https://bitbucket.org/jstrachan/environment-jx-staging-bb2/pull-requests/1/demo5-to-002/activity

maybe we can query that? It looks like Jenkins tries to update the activity on a PR /  build too:

**Comment by jstrachan** [comment]
just tried with the latest greatest code and got the same:
```
EXITCODE   0Using pipeline: jstrachan/demo6/master build: #4
error: Failed to query the Pull Request status for https://bitbucket.org/jstrachan/environment-jx-staging-bb2/pull-requests/7 Status: 404 Not Found, Body: {"type": "error", "error": {"message": "Resource not found", "detail": "There is no API hosted at this URL.\n\nFor information about our API's, please refer to the documentation at: https://developer.atlassian.com/bitbucket/api/2/reference/"}}
```
despite it now being there https://bitbucket.org/jstrachan/environment-jx-staging-bb2/pull-requests/7 

so maybe we need a bit more retry/wait in there for bitbucket?

**Comment by wbrefvem** [comment]
So currently the Bitbucket git provider returns Bitbucket-y statuses and not the generic ones that `jx promote` expects. The PR linked above addresses that.

**Comment by wbrefvem** [comment]
PR #829 seems to have fixed the status querying issue, but it seems there's now a merge issue:

```
Using pipeline: wbrefvem/create-spring-demo/master build: #7

WARNING: Failed to merge the Pull Request https://bitbucket.org/wbrefvem/environment-gemsunset-staging/pull-requests/3 due to Status: 400 Bad Request, Body: Bad Request maybe I don't have karma?
```

**Comment by ysaakpr** [comment]
[I am also facing the same issue](https://github.com/jenkins-x/jx/issues/857)

**Comment by mrinalvirnave** [comment]
I am still seeing the error. Maybe the wait is not working. 
Excerpt from Jenkins job below.
```
Created Pull Request: https://bitbucket.org/myorg/environment-jx-staging/pull-requests/1

Using pipeline: mrinal/wordcounter/master build: #6

error: Failed to query the Pull Request status for https://bitbucket.org/myorg/environment-jx-staging/pull-requests/1 Status: 404 Not Found, Body: {"type": "error", "error": {"message": "Resource not found", "detail": "There is no API hosted at this URL.\n\nFor information about our API's, please refer to the documentation at: https://developer.atlassian.com/bitbucket/api/2/reference/"}}

script returned exit code 1
```
I am at the following version
```
jx                 1.3.90
jenkins x platform 0.0.1827
kubernetes cluster v1.9.6
kubectl            v1.11.0
helm client        v2.10.0-rc.1+gaa98e7e
helm server        v2.10.0-rc.1+gaa98e7e
git                git version 2.7.4
```

**Comment by mrinalvirnave** [comment]
@wbrefvem 
I got around that by creating my gitAuth.yaml first. I think that is what worked.

**Comment by wbrefvem** [comment]
@mrinalvirnave 
That's strange, because I had already created it. Would you be willing to post your `gitAuth.yaml` so I can compare it with mine (with credentials obfuscated, of course)?

**Comment by johnmshields** [comment]
@jstrachan Please see my PR above. The PR changes have fixed this issue for me locally.

**Comment by jstrachan** [comment]
I think this issue is fixed with #1638
